core/plannernotes.c

```diff
@@ -39,15 +39,27 @@ int diveplan_duration(struct diveplan *diveplan)
  *             4) The names of the gasses in the gas change: gas-from and gas-to.
  * Returns:    The size of the output buffer that has been used after the new results have been added.
  */
-int add_icd_entry(char *icdbuffer, unsigned int maxsize, struct icd_data *icdvalues, int time_seconds, int ambientpressure_mbar, char *gasname_from, const char *gasname_to)
+int add_icd_entry(char *icdbuffer, unsigned int maxsize, struct icd_data *icdvalues, int time_seconds, int ambientpressure_mbar, struct gasmix *gas_from, struct gasmix *gas_to)
 {
-	return snprintf(icdbuffer, maxsize, 
-		"<tr><td rowspan='2' style= 'vertical-align:top;'>%3d%s</td><td rowspan=2 style= 'vertical-align:top;'>%s&#10137;%s</td><td style='padding-left: 10px;'>%+5.2f%%</td><td style= 'padding-left: 15px; color:%s;'>%+5.2f%%</td><td style='padding-left: 15px;'>%+5.2f%%</td></tr> <tr><td style='padding-left: 10px;'>%+5.2f%s</td><td style='padding-left: 15px; color:%s;'>%+5.2f%s</td><td style='padding-left: 15px;'>%+5.2f%s</td></tr>",
-		(time_seconds + 30) / 60, translate("gettextFromC", "min"), gasname_from, gasname_to, icdvalues->dHe / 10.0, 
+	int len = 0;
+	len += snprintf(icdbuffer, maxsize, 
+		"<tr><td rowspan='2' style= 'vertical-align:top;'>%3d%s</td>\
+		<td rowspan=2 style= 'vertical-align:top;'>%s&#10137;",
+		(time_seconds + 30) / 60, translate("gettextFromC", "min"), gasname(gas_from));
+	len += snprintf(icdbuffer + len, maxsize - len, 
+		"%s</td><td style='padding-left: 10px;'>%+5.2f%%</td>\
```

Comment:
I found an issue in my own code which leads to a very general question:
If I wrap around the string inside the double quotes for a printf and put the backslash at the end of the line the preprocessor puts the stuff together in one line BUT it adds the leading tabs (used for indentation). This in fact leads to an incorrect string. How can one avoid this?
@neolit123 @dirkhh @atdotde

Reply:
Ups, I guess I found the answer... much easier than expected. I will push updates for this later.